h/services/group_members.py

```diff
@@ -111,17 +115,34 @@ def update_members(self, group, userids):
         for userid in userids_for_removal:
             self.member_leave(group, userid)
 
-    def member_join(self, group, userid) -> GroupMembership:
-        """Add `userid` to the member list of `group`."""
+    def member_join(self, group, userid, roles=None) -> GroupMembership:
+        """
+        Add `userid` to `group` with `roles` and return the resulting membership.
+
+        If `roles=None` it will default to `[GroupMembershipRoles.MEMBER]`.
+
+        If a membership matching `group`, `userid` and `roles` already exists
+        in the DB it will just be returned.
+
+        :raise ConflictError: if a membership already exists with the given
+            group and userid but different roles
+        """
+        roles = roles or [GroupMembershipRoles.MEMBER]
+
         user = self.user_fetcher(userid)
 
-        existing_membership = self.get_membership(group, user)
+        kwargs = {"roles": roles}
```

Comment:
I don't think I get why we use this kwargs here vs a more explicit roles only check, are there any other membership attributes that might create a conflict coming soon?